src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,127 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static User actualUser;
+    private static User expectedUser;
+    private static RegistrationService registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.remove(actualUser);
+    }
+
+    @BeforeEach
+    void setUp() {
+        actualUser = new User();
+        expectedUser = new User();
+        actualUser.setLogin("MyName");
+        actualUser.setPassword("MyPass");
+        actualUser.setAge(18);
+        expectedUser = actualUser;
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_validLogin_Ok() {
+        actualUser.setLogin("MyName");
+        actualUser = registrationService.register(actualUser);
+        expectedUser.setLogin("MyName");
+        assertEquals(expectedUser, actualUser);

```

Comment:
And check if user has id after saving
```suggestion
    void register_Ok() {
        User actualUser = registrationService.register(expectedUser);
        assertEquals(expectedUser, actualUser);
```